mongomock/collection.py

```diff
@@ -416,6 +416,13 @@ def _update(self, spec, document, upsert=False, manipulate=False,
                     subdocument = self._update_document_fields_with_positional_awareness(
                         existing_document, v, spec, updater, subdocument)
 
+                elif k == '$rename':
+                    for src, dst in iteritems(v):
+                        if '.' in src or '.' in dst:
+                            raise NotImplementedError('renaming with dots not supported yet')
```

Comment:
Make sure to identify `mongomock` explicitly as the culprit. e.g. "Using the $rename operator with dots is a valid MongoDB operation, however it is not supported by mongomock yet".